### DISCUSSION TITLE
Why does it have to cache animation for Friction2d? Opentoonz performs way better on low end pcs but vector animation not good

### DISCUSSION THREAD

**Discussion by AlmostMikhail** [friction2d/friction#368] [Q&A]
On my laptop I only have 4 gig ram and have intel celeron cpu and no graphics card and when I run chrome browser in the background it doesn't cache entire timeline/animation so I have to close the browser and use a ram optimizer to allow timelime/animation to be cached. but, overall great software. Isn't there a way to optimize for low-end pcs?

**Comment by AlmostMikhail** [comment]
I found that if you lower resolution to 25% it caches with 4 gb ram you can go up to 2000 frames on my laptop

**Comment by rodlie** [comment]
Friction supports raster/vector/video/audio etc, so playback must be cached. Rendering requires RAM, as you noticed you can adjust the resolution to reduce the memory usage.

In the future we might add a "vector-only" playback feature that won't require caching, there are also plans for disk cache at some point.

> Isn't there a way to optimize for low-end pcs?

Friction is very optimized.

**Reply by AlmostMikhail** [reply]
https://github.com/user-attachments/assets/a84fd5e7-635c-496d-abf3-71918c582af0

Expressive Animator works without caching. I can't even render a couple of frames on my laptop with friction. How is something so simple not a feature. I might have to go to paid Expressive animator because I've been searching for an open source vector animator and I found friction 2d to be a really good software or I'll just have to wait until devs implement it

**Reply by AlmostMikhail** [reply]
Can you use Disk Caching Instead of RAM?
